torch/csrc/jit/tensorexpr/loopnest.cpp

```diff
@@ -1911,6 +1911,58 @@ std::vector<For*> LoopNest::reorder(
   return result;
 }
 
+std::vector<For*> LoopNest::tile(For* x, For* y, int x_factor, int y_factor) {
+  auto parent = dynamic_cast<Block*>(x->get_parent());
+  if (parent == nullptr) {
+    throw malformed_input("parent of the loops must be a Block");
+  }
+  if (!areLoopsPerfectlyNested({x, y})) {
+    throw malformed_input("two loops must be perfectly nested");
+  }
+
+  // Split x, y axes by x_factor and y_factor
+  For *yi, *ytail;
+  splitWithTail(y, y_factor, &yi, &ytail);
+  For *xi, *xtail;
+  splitWithTail(x, x_factor, &xi, &xtail);
+
+  // Distribute xi over yo and ytail so we can manipulate the loop order of {xo,
+  // xi, yo, yi}
+  auto loops = distributeLoop(xi);
+
+  // For {xo, xi, yo, yi}, reorder the axes to be xo, yo, xi, yi
+  xi = loops.front();
```

Comment:
This is redundant I think. After distribute, the input loop should still be the first loop.